commons/commons-logging/src/main/java/com/clicktravel/common/logging/JsonLog.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2014 Click Travel Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.clicktravel.common.logging;
+
+import org.slf4j.Logger;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Utility for logging messages as JSON rather than plain strings.
+ *
+ * Example usage:
+ *
+ * <pre>
+ * {@code
+ * private final Logger logger = LoggerFactory.getLogger(getClass());
+ *
+ * ... {
+ *
+ *     Map<String,Object> message = new HashMap<>();
+ *     Map<String,Object> userDetail = new HashMap<>();
+ *
+ *     userDetail.put("id", "9182-abf");
+ *     userDetail.put("firstName", "Bob");
+ *     message.put("eventType", "logInSuccess");
+ *     message.put("user", userDetail);
+ *
+ *     JsonLog.info(logger, message);
+ * }}
+ * </pre>
+ */
+public class JsonLog {
```

Comment:
Are we happy with no unit test coverage of this new class and it's methods?

Reply:
I've now added some unit tests